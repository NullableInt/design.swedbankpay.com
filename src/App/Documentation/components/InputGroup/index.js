import React, { Component } from "react";

import { ComponentPreview, DocToc } from "#";
import InputGroupComponent from "@/InputGroup";

const BasicExample = () => (
    <>
        <h2 id="basic-example">Basic example</h2>
        <p>Basic input group...</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                id="input-id-1"
                type="text"
                label="Name"
                placeholder="Name"
                prefixValue="Mr."
            /> {"\n\n"}
            <InputGroupComponent
                id="input-id-2"
                type="text"
                label="Email"
                placeholder="Email"
                prefixType="icon"
                prefixValue="email"
            /> {"\n\n"}
            <InputGroupComponent
                id="input-id-3"
                type="number"
                label="Amount"
                placeholder="Rounded to nearest $"
                prefixType="icon"
                prefixValue="attach_money"
                postfixValue=".00"
            /> {"\n\n"}
            <InputGroupComponent
                id="input-id-4"
                type="textarea"
                label="Notes"
                placeholder="Special requests"
                prefixType="icon"
                prefixValue="format_quote"
            />
        </ComponentPreview>
    </>
);

const CheckboxesAndRadios = () => (
    <>
        <h2 id="checkboxes-and-radios">Checkboxes and radios</h2>
        <p>Place any checkbox or radio option within an input groupâ€™s addon instead of text.</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <div className="form-group">
                <div className="input-group">{"\n"}
                    <span className="input-group-addon">{"\n\t\t"}
                        <input type="checkbox"/>{"\n\t"}
                    </span>{"\n"}
                    <input type="text" className="form-control" />{"\n"}
                </div>
            </div>{"\n\n"}
            <div className="form-group">
                <div className="input-group">{"\n"}
                    <span className="input-group-addon">{"\n\t\t"}
                        <input type="radio"/>{"\n\t"}
                    </span>{"\n"}
                    <input type="text" className="form-control" />{"\n"}
                </div>
            </div>
        </ComponentPreview>
    </>
);

const ButtonAddons = () => (
    <>
        <h2 id="button-addons">Button addons</h2>
        <p>Addons...</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                type="text"
                prefixType="button"
                prefixValue="Button"
                prefixBtnColor="secondary"
            /> {"\n\n"}
            <InputGroupComponent
                type="text"
                postfixType="button"
                postfixValue="Button"
                postfixBtnColor="primary"
            /> {"\n\n"}
            <InputGroupComponent
                type="text"
                prefixType="button"
                prefixValue="Button"
                prefixBtnColor="secondary"
                postfixType="button"
                postfixValue="Button"
                postfixBtnColor="danger"
            />
        </ComponentPreview>
    </>
);

const WithSelect = () => (
    <>
        <h2 id="with-select">With select</h2>
        <p>Select...</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                type="select"
                selectOptions={["Bounty hunter", "Smuggler", "Jedi"]}
                prefixValue="Profession"
            />
        </ComponentPreview>
    </>
);

const WithFeedbackIcon = () => (
    <>
        <h2 id="with-feedback">With feedback</h2>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                id="input-with-feedback-1"
                type="text"
                label="Name"
                placeholder="Name"
                prefixValue="Mr."
                feedbackIcon="cancel"
            />
        </ComponentPreview>
    </>
);

const ValidationStates = () => (
    <>
        <h2 id="validation-states">Validation states</h2>
        <p>Validation...</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                type="text"
                validationState="success"
                placeholder="Name"
                prefixValue="Mr."
                defaultValue="Anderson"
            /> {"\n\n"}
            <InputGroupComponent
                type="text"
                validationState="error"
                placeholder="Name"
                prefixValue="Agent"
                defaultValue="Smith"
            />
        </ComponentPreview>
    </>
);

const Disabled = () => (
    <>
        <h2 id="disabled">Disabled</h2>
        <p>Disabled blabla...</p>
        <ComponentPreview language="html" showCasePanel codeFigure removeOuterTag>
            <InputGroupComponent
                id="input-id-3"
                type="number"
                label="Amount"
                placeholder="Rounded to nearest $"
                disabled
                prefixType="icon"
                prefixValue="attach_money"
                postfixValue=".00"
            />
            <InputGroupComponent
                id="input-id-4"
                type="textarea"
                label="Special requests"
                placeholder="Extra frosting, no pineapple..."
                disabled
                prefixType="icon"
                prefixValue="format_quote"
            />
            <InputGroupComponent
                type="select"
                selectOptions={["Option one", "Option two", "Option three"]}
                label="Flavor"
                prefixValue="Options"
                disabled
            />
        </ComponentPreview>
    </>
);

const InputGroupText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <p className="lead">Input groups...</p>
        <BasicExample />
        {/* <CheckboxesAndRadios /> */}
        <ButtonAddons />
        <WithSelect />
        {/* <WithFeedbackIcon /> */}
        <ValidationStates />
        <Disabled />
    </div>
);

class InputGroup extends Component {
    componentDidMount () {
        // forms.init();
    }

    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <InputGroupText />
                    <DocToc component={InputGroupText} />
                </div>
            </div>
        );
    }
}

export default InputGroup;

/* for testing */
export { BasicExample, CheckboxesAndRadios, ButtonAddons, WithSelect, WithFeedbackIcon, ValidationStates, Disabled, InputGroupText };
