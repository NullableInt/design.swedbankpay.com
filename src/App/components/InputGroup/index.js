import React, { Fragment } from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const Addon = ({ type, value, color }) => (
    (type === "button") ?
        <div className="input-group-btn">{"\n"}
            <button type="button" className={`btn btn-${color}`}>{value}</button>{"\n"}
        </div>
        : <span className="input-group-addon">{(type === "icon") ? <i className="material-icons">{value}</i> : value}</span>
);

const Feedback = ({ icon }) => {
    return (
        <span className="form-control-feedback">{"\n\t\t"}
            <i className="material-icons">{icon}</i>{"\n\t"}
        </span>
    );
};

const InputGroup = ({
    id,
    type,
    placeholder,
    pattern,
    validate,
    required,
    defaultValue,
    autoComplete,
    disabled,
    readOnly,
    label,
    validationState,
    selectOptions,
    prefixValue,
    prefixType,
    prefixBtnColor,
    postfixValue,
    postfixType,
    postfixBtnColor,
    feedbackIcon,
    helpBlock
}) => {
    const attrs = {
        type: type || null,
        className: "form-control",
        id: id || null,
        placeholder: placeholder || null,
        defaultValue: defaultValue || "",
        disabled: disabled || null,
        readOnly: readOnly || null,
        autoComplete: autoComplete || null,
        "data-validate": validate || null,
        "data-pattern": pattern || null,
        "data-required": required || null
    };

    const inputGrpClasses = classnames(
        "input-group",
        validationState ? `has-${validationState}` : null,
        feedbackIcon ? "has-feedback" : null
    );

    return (
        <div className="form-group">{"\n"}
            {label ? <label htmlFor={id}>{label}</label> : null}{label ? "\n" : null}
            <div className={inputGrpClasses}>{"\n"}
                {prefixValue ? <Addon type={prefixType} value={prefixValue} color={prefixBtnColor} /> : null }{prefixValue ? "\n" : null}
                {type === "textarea" ?
                    <textarea {...attrs} ></textarea>
                    : type === "select" ?
                        <select className="form-control" disabled={disabled} readOnly={readOnly}>{"\n\t\t"}
                            {selectOptions.map((opt, i) => (
                                <Fragment key={opt + i}>
                                    <option>{opt}</option>{(i !== selectOptions.length - 1) ? "\n\t\t" : ""}
                                </Fragment>
                            ))}{"\n\t"}
                        </select>
                        :
                        <input {...attrs} />}
                {"\n"}
                {feedbackIcon ? <Feedback icon={feedbackIcon} /> : null} {feedbackIcon ? "\n" : null}
                {postfixValue ? <Addon type={postfixType} value={postfixValue} color={postfixBtnColor} /> : null }{postfixValue ? "\n" : null}
            </div>
            {helpBlock ? <div className="help-block">{helpBlock}</div> : null}
        </div>
    );
};

InputGroup.propTypes = {
    type: PropTypes.string.isRequired,
    id: PropTypes.string,
    placeholder: PropTypes.string,
    pattern: PropTypes.string,
    validate: PropTypes.bool,
    required: PropTypes.bool,
    defaultValue: PropTypes.string,
    autoComplete: PropTypes.string,
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    label: PropTypes.string,
    validationState: PropTypes.oneOf(["success", "error", ""]),
    selectOptions: PropTypes.array,
    prefixValue: PropTypes.string,
    prefixType: PropTypes.oneOf(["button", "icon", ""]),
    prefixBtnColor: PropTypes.oneOf(["primary", "secondary", "danger"]),
    postfixValue: PropTypes.string,
    postfixType: PropTypes.oneOf(["button", "icon", ""]),
    postfixBtnColor: PropTypes.oneOf(["primary", "secondary", "danger"]),
    feedbackIcon: PropTypes.string,
    helpBlock: PropTypes.string
};

export default InputGroup;
