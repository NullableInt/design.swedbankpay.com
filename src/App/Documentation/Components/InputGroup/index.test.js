import React from "react";
import { shallow } from "enzyme";

import InputGroup from "./index";
import { BasicExample, Demo, InputGroupText } from "./index";

describe("Documentation: InputGroup", () => {
    it("is defined", () => {
        expect(InputGroup).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <InputGroup />
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe("BasicExample", () => {
        it("is defined", () => {
            expect(BasicExample).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <BasicExample />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Demo", () => {
        it("is defined", () => {
            expect(Demo).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <Demo />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("InputGroupText", () => {
        it("is defined", () => {
            expect(InputGroupText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(
                <InputGroupText />
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
