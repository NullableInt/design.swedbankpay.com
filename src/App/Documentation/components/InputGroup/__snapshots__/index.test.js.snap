// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: InputGroup BasicExample renders 1`] = `
<Fragment>
  <h2
    id="basic-example"
  >
    Basic example
  </h2>
  <p>
    Basic input group...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <InputGroup
      id="input-id-1"
      label="Name"
      placeholder="Name"
      prefixValue="Mr."
      type="text"
    />
     
    


    <InputGroup
      id="input-id-2"
      label="Email"
      placeholder="Email"
      prefixType="icon"
      prefixValue="email"
      type="text"
    />
     
    


    <InputGroup
      id="input-id-3"
      label="Amount"
      placeholder="Rounded to nearest $"
      postfixValue=".00"
      prefixType="icon"
      prefixValue="attach_money"
      type="number"
    />
     
    


    <InputGroup
      id="input-id-4"
      label="Notes"
      placeholder="Special requests"
      prefixType="icon"
      prefixValue="format_quote"
      type="textarea"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup ButtonAddons renders 1`] = `
<Fragment>
  <h2
    id="button-addons"
  >
    Button addons
  </h2>
  <p>
    Addons...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      postfixBtnColor="primary"
      postfixType="button"
      postfixValue="Button"
      type="text"
    />
     
    


    <InputGroup
      postfixBtnColor="danger"
      postfixType="button"
      postfixValue="Button"
      prefixBtnColor="secondary"
      prefixType="button"
      prefixValue="Button"
      type="text"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup CheckboxesAndRadios renders 1`] = `
<Fragment>
  <h2
    id="checkboxes-and-radios"
  >
    Checkboxes and radios
  </h2>
  <p>
    Place any checkbox or radio option within an input groupâ€™s addon instead of text.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <div
      className="form-group"
    >
      <div
        className="input-group"
      >
        

        <span
          className="input-group-addon"
        >
          
		
          <input
            type="checkbox"
          />
          
	
        </span>
        

        <input
          className="form-control"
          type="text"
        />
        

      </div>
    </div>
    


    <div
      className="form-group"
    >
      <div
        className="input-group"
      >
        

        <span
          className="input-group-addon"
        >
          
		
          <input
            type="radio"
          />
          
	
        </span>
        

        <input
          className="form-control"
          type="text"
        />
        

      </div>
    </div>
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup Disabled renders 1`] = `
<Fragment>
  <h2
    id="disabled"
  >
    Disabled
  </h2>
  <p>
    Disabled blabla...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      disabled={true}
      id="input-id-3"
      label="Amount"
      placeholder="Rounded to nearest $"
      postfixValue=".00"
      prefixType="icon"
      prefixValue="attach_money"
      type="number"
    />
    <InputGroup
      disabled={true}
      id="input-id-4"
      label="Special requests"
      placeholder="Extra frosting, no pineapple..."
      prefixType="icon"
      prefixValue="format_quote"
      type="textarea"
    />
    <InputGroup
      disabled={true}
      label="Flavor"
      prefixValue="Options"
      selectOptions={
        Array [
          "Option one",
          "Option two",
          "Option three",
        ]
      }
      type="select"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup ValidationStates renders 1`] = `
<Fragment>
  <h2
    id="validation-states"
  >
    Validation states
  </h2>
  <p>
    Validation...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      defaultValue="Anderson"
      placeholder="Name"
      prefixValue="Mr."
      type="text"
      validationState="success"
    />
     
    


    <InputGroup
      defaultValue="Smith"
      placeholder="Name"
      prefixValue="Agent"
      type="text"
      validationState="error"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup WithFeedbackIcon renders 1`] = `
<Fragment>
  <h2
    id="with-feedback"
  >
    With feedback
  </h2>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      feedbackIcon="cancel"
      id="input-with-feedback-1"
      label="Name"
      placeholder="Name"
      prefixValue="Mr."
      type="text"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup WithSelect renders 1`] = `
<Fragment>
  <h2
    id="with-select"
  >
    With select
  </h2>
  <p>
    Select...
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    removeOuterTag={true}
    showCasePanel={true}
  >
    <InputGroup
      prefixValue="Profession"
      selectOptions={
        Array [
          "Bounty hunter",
          "Smuggler",
          "Jedi",
        ]
      }
      type="select"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: InputGroup renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Input groups...
  </p>
  <BasicExample />
  <ButtonAddons />
  <WithSelect />
  <ValidationStates />
  <Disabled />
</DocContainer>
`;
